src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs

```diff
@@ -172,19 +137,107 @@ private void WriteOperation(ClientMethod operation, CSharpNamespace? @namespace)
             }
         }
 
-        private void WriteHeaderFormat(HeaderSerializationFormat format)
+        private void WritePathSegment(PathSegment segment)
+        {
+            var value = segment.Value;
+
+            if (value.IsConstant)
+            {
+                Append("request.Uri.AppendPath(");
+                Literal(value.Constant.Value);
+                WriteSerializationFormat(segment.Format);
+                Append(", ");
+                Literal(segment.Escape);
+                Line(");");
+                return;
+            }
+
+            Append("request.Uri.AppendPath(");
+            Append(value.Parameter.Name);
+            WriteSerializationFormat(segment.Format);
+            Append(", ");
+            Literal(segment.Escape);
+            Line(");");
+        }
+
+        private void WriteHeader(RequestHeader header)
+        {
+            if (header.Value.IsConstant)
+            {
+                Append("request.Headers.Add(");
+                Literal(header.Name);
+                Append(", ");
+                Literal(header.Value.Constant.Value);
+                Line(");");
+                return;
+            }
+
+            var parameter = header.Value.Parameter;
+            var type = _typeFactory.CreateType(parameter.Type);
+            using (type.IsNullable ? If($"{parameter.Name} != null") : null)
+            {
+                Append("request.Headers.Add(");
+                Literal(header.Name);
+                Append(", ");
+                Append(parameter.Name);
+                if (type.IsNullable && type.IsValueType)
+                {
+                    Append(".Value");
+                }
+
+                WriteSerializationFormat(header.Format);
+                Line(");");
+            }
+        }
+
+        private void WriteSerializationFormat(SerializationFormat format)
         {
             var formatSpecifier = format switch
             {
-                HeaderSerializationFormat.DateTimeRFC1123 => "R",
-                HeaderSerializationFormat.DateTimeISO8601 => "S",
-                HeaderSerializationFormat.Date => "D",
+                SerializationFormat.DateTimeRFC1123 => "R",
+                SerializationFormat.DateTimeISO8601 => "S",
+                SerializationFormat.Date => "D",
+                SerializationFormat.DateTimeUnix => "U",
                 _ => null
             };
 
             if (formatSpecifier != null)
             {
-                Append($", \"{formatSpecifier}\"");
+                Append($", ");
+                Literal(formatSpecifier);
+            }
+        }
+
+        private void WriteQueryParameter(QueryParameter queryParameter)
+        {
+            ConstantOrParameter value = queryParameter.Value;
+            if (value.IsConstant)
+            {
+                Append("request.Uri.AppendQuery(");
```

Comment:
I know that the code between WriteQuery/Header/Path looks slightly duplicated but I think we need a bit of code-writer refactoring to enable deduplicating in cleanly.